File: static/solutions/java/src/main/java/com/epi/MagicMazeGame.java
package com.epi;

// @include
public class MagicMazeGame extends MazeGame {
  // The default constructor will call superclass constructor.
  @Override
  protected Room makeRoom() {
    return new MagicRoom();
  }
}
//@exclude

class MagicRoom implements Room {
  public void connect(Room that) {}
}

File: solutions/python/magic_maze_game.py
from maze_game import MazeGame
from maze_game import Room

# @include
class MagicMazeGame(MazeGame):

    def make_room(self):
        return MagicRoom()


class MagicRoom(Room):

    def connect(self, room2):
        print("Connecting magic room")
# @exclude

x = MagicMazeGame()
assert x is not None
assert x.make_room() is not None
